pytorch_lightning/strategies/deepspeed.py

```diff
@@ -651,7 +653,7 @@ def _auto_select_batch_size(self):
 
     def _format_precision_config(self) -> None:
         if self.precision_plugin.precision in (PrecisionType.HALF, PrecisionType.MIXED):
-            if "fp16" not in self.config and self.precision_plugin.amp_type == AMPType.NATIVE:
+            if "fp16" not in self.config and isinstance(self.precision_plugin, NativeMixedPrecisionPlugin):
```

Comment:
Note, the test (test_trainer_model_hook_system_fit) failed because the type was actually `DeepSpeedPrecisionPlugin` which also has a field amp_type (set to NATIVE).
The same would be true for the apex check in the "elif" block below. 
Both if statements will be skipped with the changes in this PR. 

Since the DeepSpeedPrecisionPlugin is a weird basically bundling both native and apex together, we probably have to revert the changes here or have a more radical change.

Reply:
Deepspeed only supports native AMP, right?

Reply:
It appears so yes. I too think it would be worth investigating  #12323